test/exportsField.js

```diff
@@ -923,3 +929,87 @@ describe("Exports field", function exportsField() {
 		});
 	});
 });
+
+describe("ExportsFieldPlugin", () => {
+	const nodeFileSystem = new CachedInputFileSystem(fs, 4000);
+
+	const resolver = ResolverFactory.createResolver({
+		extensions: [".js"],
+		fileSystem: nodeFileSystem,
+		conditionNames: ["webpack"]
+	});
+
+	it("resolve root using exports field, not a main field", done => {
+		resolver.resolve({}, fixture, ".", {}, (err, result) => {
+			if (err) return done(err);
+			if (!result) throw new Error("No result");
+			result.should.equal(path.resolve(fixture, "index.js"));
+			done();
+		});
+	});
+
+	it("resolve using exports field, not a browser field", done => {
+		const resolver = ResolverFactory.createResolver({
+			aliasFields: ["browser"],
+			extensions: [".js"],
+			fileSystem: nodeFileSystem,
+			conditionNames: ["webpack"]
+		});
+
+		resolver.resolve({}, fixture, "./lib/main.js", {}, (err, result) => {
+			if (err) return done(err);
+			if (!result) throw new Error("No result");
+			result.should.equal(path.resolve(fixture, "./lib/lib2/main.js"));
+			done();
+		});
+	});
+
+	it("throw error if extension not provided", done => {
+		resolver.resolve({}, fixture, "./lib/main", {}, (err, result) => {
```

Comment:
As far as I understood the behavior these requests should not be affected by the exports field. Exports field should only applied for requests from outside: So only `exports-field/lib/main.js` but not `./lib/main.js` inside of `exports-field`.

Reply:
Sure, I forgot about it..

Reply:
> Exports field should only applied for requests from outside

There is also self-referencing, see test-cases